NethermindNode.Tests/Tests/SyncingNode/BodiesAndReceiptsFinalization.cs

```diff
@@ -0,0 +1,108 @@
+
+            // 1
+            NodeInfo.WaitForNodeToBeReady(Logger);
+
+            var delay = Backoff.ConstantBackoff(TimeSpan.FromSeconds(3), retryCount: 100);
+
+            var retryPolicy = Policy
+            .HandleResult<string>(s => !s.Contains("SnapSync") && !s.Contains("StateNodes"))
+            .WaitAndRetry(delay);
+
+            string result = retryPolicy.Execute(() => NodeInfo.GetCurrentStage(Logger));
+
+            NodeInfo.WaitForNodeToBeSynced(Logger);
+
+            // 2
+            DockerCommands.StopDockerContainer(ConfigurationHelper.Instance["execution-container-name"], Logger);
+
+            // 3
+            var execPath = DockerCommands.GetExecutionDataPath(Logger);
+            CommandExecutor.BackupDirectory(execPath + "/nethermind_db", execPath + "/nethermind_db_backup" , Logger);
+
+            // 4
+            string[] flagsToRemove =
+            {
+                "--Sync.NonValidatorNode=true",
+                "--Sync.DownloadBodiesInFastSync=false",
+                "--Sync.DownloadReceiptsInFastSync=false"
+            };
+
+            Logger.Info("Reading docker compose file at path: " + execPath + "/../docker-compose.yml");
+            var dockerCompose = DockerComposeHelper.ReadDockerCompose(execPath + "/../docker-compose.yml");
+            foreach (var flag in flagsToRemove)
+            {
+                Console.WriteLine("Removing: " + flag);
+                DockerComposeHelper.RemoveCommandFlag(dockerCompose, "execution", flag);
+            }
+            DockerComposeHelper.WriteDockerCompose(dockerCompose, execPath + "/../docker-compose.yml");
+            DockerCommands.RecreateDockerCompose("execution", execPath + "/../docker-compose.yml", Logger);
+            DockerCommands.StartDockerContainer(ConfigurationHelper.Instance["execution-container-name"], Logger);
+
+            // 5-6-7
+            for (int i = 0; i < repeatCount; i++)
+            {
+                NodeInfo.WaitForNodeToBeReady(Logger);
+                NodeInfo.WaitForNodeToBeSynced(Logger);
+                var bodiesLine = DockerCommands.GetDockerLogs(ConfigurationHelper.Instance["execution-container-name"], "Fast blocks bodies task completed.");
+                var receiptsLine = DockerCommands.GetDockerLogs(ConfigurationHelper.Instance["execution-container-name"], "Fast blocks receipts task completed.");
+
+                Assert.IsTrue(bodiesLine.Count() > 0, "Bodies log line missing - verify with getBlockByNumber.");
+                Assert.IsTrue(receiptsLine.Count() > 0, "Receipts log line missing - verify with getReceipt");
+
+                DockerCommands.StopDockerContainer(ConfigurationHelper.Instance["execution-container-name"], Logger);
+
+                CommandExecutor.RemoveDirectory(execPath + "/nethermind_db", Logger);
+                CommandExecutor.BackupDirectory(execPath + "/nethermind_db_backup", execPath + "/nethermind_db", Logger);
```

Comment:
Probably `BackupDirectory` is not the best name. Maybe `CopyDirectory` will work better?